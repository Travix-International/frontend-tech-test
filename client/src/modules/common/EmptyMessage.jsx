import React from 'react';
import PropTypes from 'prop-types';

export default class EmptyMessage extends React.Component {
  static propTypes = {
    isEmpty: PropTypes.bool.isRequired,
    renderMessage: PropTypes.func.isRequired,
    className: PropTypes.string,
    children: PropTypes.any.isRequired,
  }

  static defaultProps = {
    className: '',
  }

  get emptyView() {
    const { renderMessage } = this.props;

    return (
      <div>
        <div className="EmptyMessage-Sign">
          Here goes an empty sign
        </div>
        <div className="EmptyMessage-Message">
          {renderMessage()}
        </div>
      </div>
    );
  }

  render() {
    const { isEmpty, children } = this.props;
    return (
      <div className={`EmptyMessage ${this.props.className}`}>
        {isEmpty ? this.emptyView : children}
      </div>
    );
  }
}
