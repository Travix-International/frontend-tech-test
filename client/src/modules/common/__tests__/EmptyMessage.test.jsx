import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import EmptyMessage from '../EmptyMessage';

const renderMessage = () => <div className="message" />;

describe('Common.EmptyMessage', () => {
  it('should render with props', () => {
    const isEmpty = true;
    const children = <div className="children" />;

    const emptyMessage = shallow(
      <EmptyMessage isEmpty={isEmpty} renderMessage={renderMessage}>
        {children}
      </EmptyMessage>,
    );

    expect(toJson(emptyMessage)).toMatchSnapshot();
  });

  it('should render children when isEmpty not true', () => {
    const isEmpty = false;
    const children = <div className="children" />;

    const emptyMessage = shallow(
      <EmptyMessage isEmpty={isEmpty} renderMessage={renderMessage}>
        {children}
      </EmptyMessage>,
    );

    expect(emptyMessage.contains(children)).toBe(true);
  });

  it('should render a message with renderMessage when isEmpty === true', () => {
    const isEmpty = true;
    const children = <div className="children" />;

    const emptyMessage = shallow(
      <EmptyMessage isEmpty={isEmpty} renderMessage={renderMessage}>
        {children}
      </EmptyMessage>,
    );

    expect(emptyMessage.find('.message').length).toBe(1);
  });
});
