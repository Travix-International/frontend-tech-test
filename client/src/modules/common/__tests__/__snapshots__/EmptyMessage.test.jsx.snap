// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common.EmptyMessage should render with props 1`] = `
<div
  className="EmptyMessage "
>
  <div>
    <div
      className="EmptyMessage-Sign"
    >
      Here goes an empty sign
    </div>
    <div
      className="EmptyMessage-Message"
    >
      <div
        className="message"
      />
    </div>
  </div>
</div>
`;
